frontend/src/app/components/ck-monitor/ck-monitor.component.ts

```diff
@@ -260,44 +313,86 @@ export class CkMonitorComponent implements OnInit, OnDestroy {
     });
   }
 
-  async view(task: TaskWorkflow): Promise<void> {
+  async view(task: TaskWorkflow, status: GroupTaskStatus): Promise<void> {
     this.runningTask = task;
+    this.runningTaskGroupStatus = status;
     const progressData = await this._calcAverageProgress(
       this.taskWorkflowGroupMap.get(this.runningTask)
     );
     this.minGroupProgress = progressData[0];
     this.averageGroupProgress = progressData[1];
     this.maxGroupProgress = progressData[2];
-    await this.updateRunningTaskDataSource(task);
+    await this.updateRunningTaskDataSource(task, status);
     this._startListening();
   }
 
-  async updateRunningTaskDataSource(workflow: TaskWorkflow): Promise<void> {
+  async updateRunningTaskDataSource(
+    workflow: TaskWorkflow,
+    status: GroupTaskStatus
+  ): Promise<void> {
     const groupTasks = this.taskWorkflowGroupMap.get(workflow);
     const groupTasksTableFormat: MonitorData[] = [];
     if (groupTasks) {
       for (let i = 0; i < groupTasks.length; i++) {
-        const groupMembers: string[] = [];
-        for (let j = 0; j < groupTasks[i].group.members.length; j++) {
-          groupMembers.push(
-            (await this.userService.getOneById(groupTasks[i].group.members[j]))
-              .username
-          );
+        if (groupTasks[i].groupTask.status == status) {
```

Comment:
I think this looks fine, but was wondering if we could just embed the workspace angular component into the ck monitor? This would lower the duplicate code right? Or is this different from the workspace?